apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,406 @@
+
+## Pick your poison
+
+This article covers configuration of all possible combinations in modern Node.js (v12+). If you are trying to decide which options are ideal, it is better to avoid dual packages, so either:
+
+- ESM source and distribution
+- CJS source and distribution with good/specific `module.exports`
+
+| You as a package author write     | Consumers of your package write their code in                    | Your options                                                                                |
+| :-------------------------------- | :--------------------------------------------------------------- | :------------------------------------------------------------------------------------------ |
+| CJS source code using `require()` | CJS: consumers `require()` your package                          | [CJS source and distribution](#cjs-source-and-distribution)                                 |
+| CJS source code using `require()` | ESM: consumers `import` your package                             | [CJS source and only ESM distribution](#cjs-source-and-only-esm-distribution)               |
+| CJS source code using `require()` | CJS & ESM: consumers either `require()` or `import` your package | [CJS source and both CJS & ESM distribution](#cjs-source-and-both-cjs-amp-esm-distribution) |
+| ESM source code using `import`    | CJS: consumers `require()` your package                          | [ESM source with only CJS distribution](#esm-source-with-only-cjs-distribution)             |
+| ESM source code using `import`    | ESM: consumers `import` your package                             | [ESM source and distribution](#esm-source-and-distribution)                                 |
+| ESM: source code uses `import`    | CJS & ESM: consumers either `require()` or `import` your package | [ESM source and both CJS & ESM distribution](#esm-source-and-both-cjs-amp-esm-distribution) |
+
+### CJS source and distribution
+
+This the "Rum & Coke" of packages: pretty difficult to mess up. Essentially just declare the package’s exports via the [`"exports"`](https://nodejs.org/api/packages.html#conditional-exports) field/field-set.
+
+**Working example**: [cjs-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/cjs-distro)
+
+```json
+{
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" }, // optional, but kind
+  "exports": {
```

Comment:
Is this necessary? At most it can use the type field, but AFAICT you just need to point the main field and don't need to bother with exports.

Reply:
Maybe technically yes. But this guide is about _should_ not necessarily _must_.

Reply:
That may be true before it's moved to the website, but now it's in the learn session I think we should target the audience differently (which I think was the core of the doc/ambassador program in the past two summits): we should teach more best practices that are approachable to beginners and up-to-date. Having the guide about package shipping very pro-configuration when it's not necessary and is not what the majority of the ecosystem does is deviating from that goal IMO.

Reply:
How about 2 versions of each: "basic" and "advanced"?